src/io.c

```diff
@@ -137,6 +137,57 @@ static int pids[MAXCHLDS];         /* and this number! */
 static int fistats = 0;            /* First used entry */
 static int lastats = 0;            /* First unused entry */
 static int statsfull = 0;          /* Flag, whether stats FIFO full */
+
+
+// This function must only be called while IO's signal handler is disabled!
+static int findSignaledPid(int pidc)
+{
+    if (fistats == lastats && !statsfull) /* queue empty */
+        return -1;
+
+    int pos = fistats;
+    while (pids[pos] != pidc)
+    {
+        pos++;
+        if (pos >= maxstats) pos = 0;
+        if (pos == lastats) {
+            pos = -1;  /* None found */
+            break;
+        }
+    }
+    return pos;
+}
+
+// This function must only be called while IO's signal handler is disabled!
+static void removeSignaledPidByPos(int pos)
+{
+    if (fistats == lastats && !statsfull) /* queue empty */
+        return;
+
+    int newpos;
+    if (pos == fistats) {  /* this is the easy case: */
+        fistats++;
+        if (fistats >= maxstats) fistats = 0;
+    } else {  /* The more difficult case: */
+        do {
+            newpos = pos+1;
+            if (newpos >= maxstats) newpos = 0;
+            if (newpos == lastats) break;
+            stats[pos] = stats[newpos];
+            pids[pos] = pids[newpos];
+            pos = newpos;
+        } while(1);
+        lastats = pos;
+    }
+    statsfull = 0;
+}
+
+/* This does not have to be the same size as the array above */
```

Comment:
So... why do you mention it? Why would one think it has to be? I am confused :-)

Reply:
Because I am using MAXCHLDS (because it seems like a sensible value), but someone might think both arrays have to be MAXCHLDS.